### DISCUSSION TITLE
OTA update

### DISCUSSION THREAD

**Discussion by d-solis-tovar** [micropython/micropython#13120] [ESP32] [answered]
Hi,

I want to implement OTA updates for my ESP32 microcontroller. I flashed micropython firmware with OTA support (one partition for `ota_0` and another partition for `ota_1`). I'm looking for a procedure like this when running partition is `ota_0`:

  1. Get new firmware `.bin` from a request to Github or any other server (I created a custom Micropython firmware)
  2. Write `.bin` content into partition `ota_1`
  3. Set partition `ota_1` to run at next boot

Right now, I'm stuck trying to at Step 1. I tried to get/download .bin file using `urequests.get` but it didn't work.  

`.bin` file is 1.6 MB size.

Thank you in advance

**Comment by glenn20** [comment, marked as answer]
You should use the `.app_bin` files (which can be downloaded from from https://micropython.org/download/?mcu=esp32), not the micropython `.bin` files, for OTA updates.

See https://github.com/glenn20/micropython-esp32-ota for more details abour the `.app-bin` files and examples (and a tool to perform ota updates from urls).

**Reply by ThinkTransit** [reply]
@glenn20 That is a great project for OTA, thanks for creating!
